### PR TITLE
Fix potential segfault in memcmp64

### PR DESCRIPTION
**Pull request by amosr**
It was reading past the end of buffers, which could have caused a segfault one day

### CODE REVIEW

**Comment by amosr** [comment]
! @tranma @jystic

File: csrc/anemone_memcmp.h [hunk 1]
```diff
@@ -105,29 +105,42 @@ int anemone_memcmp64 (const void *as, const void *bs, size_t len)
...
-        }
-        return diff2;
-    */
+    uint8_t *a8 = (uint8_t*)as;
+    uint8_t *b8 = (uint8_t*)bs;
+
+    int diff;
+
+    switch (rem) {
+    case 7:
```
Comment (erikd-ambiata) [inline csrc/anemone_memcmp.h]: A bit like Duff's technique. Nice!

**Comment by jystic** [comment]
👍

File: cbench/bench_memcmp.c [hunk 2]
```diff
@@ -19,36 +19,73 @@ const char* rubbish
    "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
    "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
 
-#define NUM_BENCHES 1000000
-#define MKBENCH(f)                                                  \
-int64_t f##_bench (uint64_t len)                                    \
+#define NUM_BENCHES 10000000
```
Comment (tranma) [inline cbench/bench_memcmp.c]: verified performance

**Pull request merged**